R/plot.R

```diff
@@ -181,6 +181,33 @@ plot.tidyvpcobj <- function(x,
   g
 }
 
+#' Expand single-value vpc groups to a finite width so that they show up with `geom_ribbon()`
+#'
+#' @param vpc The vpc object
+#' @return A data frame of the vpc$stats possibly with additional rows for
+#'   single-value groups
+#' @noRd
+expand_vpc_stats_single_value <- function(vpc, xvar, width = 0.0001) {
+  d_vpc_stats <- vpc$stats
+  if (!is.null(vpc$strat)) {
+    d_vpc_stats[, n_xvar := length(unique(get(xvar))), by = names(vpc$strat)]
```

Comment:
Note that with data.table, we can modify `d_vpc_stats` directly, adding or updating the n_xvar column, without needing to reassign it with `d_vpc_stats <- ...`